conans/client/build/cmake.py

```diff
@@ -118,15 +118,42 @@ def is_multi_configuration(self):
 
     @property
     def command_line(self):
-        args = ['-G "%s"' % self.generator] if self.generator else []
+        if self.generator_platform and not is_generator_platform_supported(self.generator):
+            raise ConanException('CMake does not support generator platform with generator '
+                                 '"%s:. Please check your conan profile to either remove the '
+                                 'generator platform, or change the CMake generator.'
+                                 % self.generator)
+
+        if self.toolset and not is_toolset_supported(self.generator):
+            raise ConanException('CMake does not support toolsets with generator "%s:.'
+                                 'Please check your conan profile to either remove the toolset,'
+                                 ' or change the CMake generator.' % self.generator)
+
+        generator = self.generator
+        generator_platform = None
+
         if self.generator_platform:
-            if is_generator_platform_supported(self.generator):
-                args.append('-A "%s"' % self.generator_platform)
+            if self.generator == "Visual Studio 16 2019":
```

Comment:
Maybe this whole logic can be simplified. If ``self.generator_platform`` is None, it can be assigned to ``generator_platform`` anyway. Also, structure the cases top-down: first check if Visual Studio, if true, check if VS16 or WindowsCE, etc.